src/pages/permissions/panels/IdentityGroupsPanelConfirmModal.tsx

```diff
@@ -0,0 +1,108 @@
+import { ConfirmationModal, useNotify } from "@canonical/react-components";
+import { FC, useState } from "react";
+import { LxdIdentity } from "types/permissions";
+import {
+  generateGroupAllocationsForIdentities,
+  getChangesInGroupsForIdentities,
+  pivotIdentityGroupsChangeSummary,
+} from "util/permissionIdentities";
+import GroupsOrIdentityChangesTable from "./GroupOrIdentityChangesTable";
+import { updateIdentities } from "api/auth-identities";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { useToastNotification } from "context/toastNotificationProvider";
+import usePanelParams from "util/usePanelParams";
+
+interface Props {
+  onConfirm: () => void;
+  close: () => void;
+  selectedIdentities: LxdIdentity[];
+  addedGroups: Set<string>;
+  removedGroups: Set<string>;
+}
+
+const IdentityGroupsPanelConfirmModal: FC<Props> = ({
+  onConfirm,
+  close,
+  addedGroups,
+  removedGroups,
+  selectedIdentities,
+}) => {
+  const [submitting, setSubmitting] = useState(false);
+  const notify = useNotify();
+  const panelParams = usePanelParams();
+  const queryClient = useQueryClient();
+  const toastNotify = useToastNotification();
+
+  const identityGroupsChangeSummary = getChangesInGroupsForIdentities(
+    selectedIdentities,
+    addedGroups,
+    removedGroups,
+  );
+
+  const groupIdentitiesChangeSummary = pivotIdentityGroupsChangeSummary(
+    identityGroupsChangeSummary,
+  );
+
+  const handleSaveGroupsForIdentities = () => {
+    setSubmitting(true);
+
+    const newGroupsForIdentities = generateGroupAllocationsForIdentities(
+      addedGroups,
+      removedGroups,
+      selectedIdentities,
+    );
+
+    const payload = selectedIdentities.map((identity) => ({
+      ...identity,
+      groups: newGroupsForIdentities[identity.id],
+    }));
+
+    updateIdentities(payload)
+      .then(() => {
+        // modifying groups should invalidate both identities and groups api queries
+        void queryClient.invalidateQueries({
+          predicate: (query) => {
+            return [queryKeys.identities, queryKeys.authGroups].includes(
+              query.queryKey[0] as string,
+            );
+          },
+        });
+        const successMessage =
+          selectedIdentities.length > 1
+            ? `Updated groups for ${selectedIdentities.length} identities`
+            : `Updated groups for ${selectedIdentities[0].name}`;
```

Comment:
This should report on the effective size. The selected identities count is not always matching the actually changed identities.
This is also broken in `GroupIdentitiesPanelConfirmModal`.